src/core/segment_reader.rs

```diff
@@ -280,6 +286,117 @@ impl SegmentReader {
         Ok(inv_idx_reader)
     }
 
+    /// Returns the list of fields that have been indexed in the segment.
+    /// The field list includes the field defined in the schema as well as the fields
+    /// that have been indexed as a part of a JSON field.
+    /// The returned field name is the full field name, including the name of the JSON field.
+    ///
+    /// The returned field names can be used in queries.
+    ///
+    /// Notice: If your data contains JSON fields this is **very expensive**.
```

Comment:
/// Disclaimer
///
/// Some fields may not be listed here. For instance, if the schema contains a json field that is not
// indexed nor a fast field but is stored, it is possible for the field to not be listed.